pandas/_libs/window.pyx

```diff
@@ -1754,6 +1754,135 @@ def _roll_weighted_sum_mean(float64_t[:] values, float64_t[:] weights,
     return output
 
 
+# ----------------------------------------------------------------------
+# Rolling var for weighted window
+
+
+cdef inline float64_t calc_weighted_var(float64_t t,
+                                        float64_t sum_w,
+                                        Py_ssize_t win,
+                                        int ddof,
+                                        float64_t nobs,
+                                        int64_t minp) nogil:
+    cdef:
+        float64_t result
+
+    # Variance is unchanged if no observation is added or removed
+    if (nobs >= minp) and (nobs > ddof):
+
+        # pathological case
+        if nobs == 1:
+            result = 0
+        else:
+            result = t * win / ((win - <float64_t>ddof) * sum_w)
+            if result < 0:
```

Comment:
Would this only be negative if the weight was negative which we should validate much earlier?

Reply:
I guess it might be if one weight is negative and another is positive but not necessarily